### ISSUE TITLE
Agent should monitor free disk space

### ISSUE THREAD

**Issue by AnarManafov** [FairRootGroup/DDS#392]
During ALICE tests there are many cases when agents are started on nodes with low disk space. It might lead to an abort as the agent is not even able to log or execute other disk related operations.

Introduce a new configuration option to control the minimum free space. Once the defined threshold is reached, the agent should log a fatal error and shutdown user tasks and itself.

**Comment by AnarManafov** [comment]
The agent now constantly monitors the available disk space.
It reports the available space at start
~~~~
2021-10-14 14:06:50.579038   inf    dds-agent            <0x00010dbd:0x000000011b63fe00>    Avaliable disk space: 295.6GB (317310652416)
~~~~
Then it schedules this check task every 30 seconds.

If the scheduled task detects that the disk space is lower then the defined threshold, then it will log the following message and trigger a self-shutdown.
~~~~
Triggering a shutdown due to low disk space. Required: 300GB (322122547200) Available: 295.6GB (317309726720)
~~~~

Now I will look for a suitable way to report this event to users via the Tools API.

**Comment by AnarManafov** [comment]
The error propagation will be followed up by GH-351
